protected/modules/Genetics/models/GeneticsPatient.php

```
@@ -213,42 +213,42 @@ protected function afterSave()
 
         if($this->getIsNewRecord()) {
             $this->updateDiagnoses();
+        }
 
-            /*
-             * Auto-generate a new pedigree and
-             * assign it to the genetic subject
-             */
-
-            if(isset($_POST['no_pedigree']))
-            {
-                $pedigree_inheritance = PedigreeInheritance::model()->findByAttributes(array('name' => 'Unknown/other'));
+        /*
+         * Auto-generate a new pedigree and
+         * assign it to the genetic subject
+         */
 
-                $pedigree = new Pedigree();
+        if(isset($_POST['no_pedigree']))
+        {
+            $pedigree_inheritance = PedigreeInheritance::model()->findByAttributes(array('name' => 'Unknown/other'));
```

This is quite brittle, can we raise a ticket to use an attribute to define the unknown inheritance model ... if the name is changed, this will fail.